golem-component-service-base/src/service/component.rs

```diff
@@ -152,6 +233,19 @@ pub trait ComponentService<Namespace> {
         component_name: &ComponentName,
         component_type: ComponentType,
         data: Vec<u8>,
+        files: Option<InitialComponentFilesArchiveAndPermissions>,
+        namespace: &Namespace,
+    ) -> Result<Component<Namespace>, ComponentError>;
+
+    // Files must have been uploaded to the blob store before calling this method
+    async fn create_internal(
```

Comment:
Could you quickly explain the difference here with `create`. I feel this is slightly a complicated workflow.
Can't `create` handle the situation of making sure the files are already in blob-store?

Reply:
The problem is that I want to accept both an archive and a list of paths in create.

In case of internal create I want to take the files from blob storage (meaning that I need to be able to set the archive to None, to avoid a useless update). On the other hand for create through the rest api I always want to have either both an archive and a list or neither, so I don't really want to have two separate optional parameters here.

This way was the best I could come up with.